docs/handbook/operation_order.md

```diff
@@ -0,0 +1,110 @@
+# Environment operation order
+
+This chapter attempts to document the order of important operations done during environment transitions.
+Since AliECS is an evolving system, the information presented here might be out-of-date, thus please refer to event handling in [core/environment/environment.go](https://github.com/AliceO2Group/Control/blob/master/core/environment/environment.go) and plugin calls in [ControlWorkflows/workflows/readout-dataflow.yaml](https://github.com/AliceO2Group/ControlWorkflows/blob/master/workflows/readout-dataflow.yaml) for the ultimate source of truth.
+Also, please report to the ECS developers any inaccuracies.
+
+[State Machine Callbacks](configuration.md#State-machine-callbacks) documents the order of callbacks that can be associated with state machine transitions.
+
+## START_ACTIVITY (Start Of Run)
+
+This is the order of actions happening at a healthy start of run.
+
+### before_START_ACTIVITY
+
+- `"run_number"` is set.
+- `"run_start_time_ms"` is set using the current time. It is considered as the SOR and SOSOR timestamps.
+- `before_START_ACTIVITY` hooks are executed:
+  - `trg.PrepareForRun()` at `-200`
+  - `trg.RunLoad()`, `bookkeeping.StartOfRun()` at `-100`
+  - `bookkeeping.RetrieveFillInfo()` at `-99`
+  - `kafka.PublishStartActivityUpdate()` at `-50`
+  - `dcs.StartOfRun()`, `odc.Start()` (does not need to return now), `ccdb.RunStart()` at `0`
+
+### before_event
+
+- `before_event` hooks are executed
+
```

Comment:
Thank you for starting this. I've double checked now and it looks like we do not support generic targets like `before_event` or `leave_state`, all the targets must be explicit like `before_START_ACTIVITY` and `leave_CONFIGURED`. The order reported in [configuration.md](configuration.md#State-machine-callbacks) refers to the documentation of the FSM package, but we've since implemented our own callback management. The order reported there is still correct, except the generic targets are not supported.

Reply:
Thanks, I wasn't aware. I will update the doc.